src/timesync/timesyncd-conf.c

```diff
@@ -36,6 +37,16 @@ int manager_parse_server_string(Manager *m, ServerType type, const char *string)
                 if (r == 0)
                         break;
 
+                r = dns_name_is_valid_or_address(word);
+                if (r < 0) {
+                        log_error_errno(r, "Failed to check validity of NTP server name or address '%s', ignoring: %m", word);
```

Comment:
This should return the error, like for `extract_first_word` above. I'm assuming that an error is only a resource error, like oom.